app/src/main/java/org/p2p/wallet/infrastructure/network/data/RpcTransactionError.kt

```diff
@@ -0,0 +1,64 @@
+package org.p2p.wallet.infrastructure.network.data
+
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import org.p2p.wallet.utils.getOrNull
+
+/**
+ * For parsing structs like:
+ * ```json
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        { "Custom": 6022 }
+ *    ]
+ * }
+ *
+ *    or
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        "AccountInUse"
+ *    ]
+ * }
+ * ```
+ * @return valid typed RpcTransactionError or null if parsing is failed
+ */
+sealed interface RpcTransactionError {
+    companion object {
+        fun from(errJson: JsonObject): RpcTransactionError? {
+            // for example "InstructionError"
+            val transactionErrorType: String? = errJson.keySet().firstOrNull()
+            return when (transactionErrorType) {
+                "AccountInUse" -> AccountInUse
+                "InstructionError" -> parseInstructionError(errJson, transactionErrorType)
+                else -> null
+            }
+        }
+    }
+
+    object AccountInUse : RpcTransactionError
```

Comment:
can AccountInUse and InstructionError be inherited from RpcTransactionError sealed class ?

Reply:
already inherited, or i haven't catched your question